src/components/AttachmentPreview.tsx

```diff
@@ -0,0 +1,113 @@
+import {Str} from 'expensify-common';
+import {ResizeMode, Video} from 'expo-av';
+import React, {useMemo} from 'react';
+import {View} from 'react-native';
+import useThemeStyles from '@hooks/useThemeStyles';
+import variables from '@styles/variables';
+import DefaultAttachmentView from './Attachments/AttachmentView/DefaultAttachmentView';
+import Icon from './Icon';
+import * as Expensicons from './Icon/Expensicons';
+import ImageView from './ImageView';
+import {PressableWithFeedback} from './Pressable';
+
+type ImagePickerResponse = {
+    height?: number;
+    name: string;
+    size?: number | null;
+    type: string;
+    uri: string;
+    width?: number;
+};
+
+type FileObject = Partial<File | ImagePickerResponse>;
+
+type AttachmentPreviewProps = {
+    /** Source (URL, SVG function) for file. */
+    source: string;
+
+    /** Media's aspect ratio to calculate the thumbnail */
+    aspectRatio: number | undefined;
+
+    /** Function to call when pressing thumbnail */
+    onPress: () => void;
+};
+
+function AttachmentPreview({source, aspectRatio = 1, onPress}: AttachmentPreviewProps) {
+    const styles = useThemeStyles();
+
+    const file = useMemo<FileObject | undefined>(() => {
+        const originalFileName: string = source.split('/').pop() ?? '';
+        return originalFileName
+            ? {
+                  name: originalFileName,
+              }
+            : undefined;
+    }, [source]);
+
+    const isSourceImage = typeof source === 'number' || (typeof source === 'string' && Str.isImage(source));
+    const isSourceVideo = ((typeof source === 'string' && Str.isVideo(source)) || (file?.name && Str.isVideo(file.name))) ?? (file?.name && Str.isVideo(file.name));
+    const isFileNameImage = file?.name && Str.isImage(file.name);
+    const isFileImage = isSourceImage || isFileNameImage;
+    const isFileVideo = isSourceVideo && typeof source === 'string';
```

Comment:
This looks quite cryptic, I think it would be nice to move it to utils. A function with just one or two arguments that return all the booleans. I think it could be also reused in other parts of the App.